azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -311,7 +305,7 @@ def should_enforce_signature_validation(self, extension_to_check):
 
         global_signature_required = self._policy.get("extensionPolicies").get("signatureRequired")
         individual_policy = self._policy.get("extensionPolicies").get("extensions").get(extension_to_check.name)
-        if individual_policy is None:
+        if individual_policy is None or len(individual_policy) == 0:
```

Comment:
what if individual policy has 'runtimePolicy' attribute, but not 'signatureRequired' attribute?
Then we enter else condition and return None. 

I think instead we should check
>> if individual_policy is None or individual_policy.get('signatureRequired') is None

and add unit test

Reply:
whoa, good catch!

I could try to explain why I missed this, but that would sound just like an excuse for my laziness. Updated code and added unit test.

@mgunnala - I refactored some of the unit tests as part of this